Files/Integrity/IChecksum.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Files
+{
+    public interface IChecksum
+    {
+        public void Update(IEnumerable<byte> buffer);
```

Comment:
why not `byte[]` but `IEnumerable<byte>`?

Reply:
different hashing APIs take different things, and we're going to pass an `IEnumerable<byte>` anyways (due to using `Take(int)` on a `byte[]`), and it's just easier this way, because we can handle the type conversions in the implementing classes instead.